payload/Library/Scripts/install_or_defer.sh

```diff
@@ -332,6 +332,17 @@ if [[ $? -ne 0 ]]; then
     BAILOUT=true
 fi
 
+# Check if a custom CatalogURL is set and if it is available
```

Comment:
It may make sense to put this as an `else` for the above ping check. In other words: if the ping fails, we don't bother trying to `curl` the CatalogURL. This would simplify the resulting error output.